### PR TITLE
[RDY] API: fix nvim_command_output buffer overflow

### LINKED ISSUE #10829: API: nvim_command_output buffer overflow
- `nvim --version` ([master](https://github.com/neovim/neovim/tree/ed28668392e450bf22fe97e0581a63135498b8c0)):
```
NVIM v0.4.0-1791-ged2866839
Build type: Debug
LuaJIT 2.0.5
Compilation: /usr/bin/clang -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fdiagnostics-color=always -fno-sanitize-recover=all -fno-omit-frame-pointer -fno-optimize-sibling-calls -fsanitize=address -fsanitize=undefined -fsanitize-blacklist=/home/abdelhakeem/neovim/src/.asan-blacklist -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DEXITFREE -I/home/abdelhakeem/neovim/asan-build/config -I/home/abdelhakeem/neovim/src -I/home/abdelhakeem/neovim/.deps/usr/include -I/usr/include -I/home/abdelhakeem/neovim/asan-build/src/nvim/auto -I/home/abdelhakeem/neovim/asan-build/include
Compiled by abdelhakeem@abdelhakeem

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/local/share/nvim"

Run :checkhealth for more info
```
- Operating system/version: `Linux 5.1.15-arch1-1-ARCH #1 SMP PREEMPT Tue Jun 25 04:49:39 UTC 2019 x86_64 GNU/Linux`
- Terminal name/version: `# GNOME Terminal 3.32.2 using VTE 0.56.3 +GNUTLS`
- `$TERM`: `xterm-256color`

### Steps to reproduce using `nvim -u NORC` (ASAN build)

```
nvim -u NORC
:call nvim_command_output('echo range(1000000)')
```

### Actual behaviour

Heap buffer overflow.

### Expected behaviour

Return command output.

### PR DESCRIPTION
**Pull request by abdelhakeem**
closes #10829

### CODE REVIEW

**Review by justinmk — APPROVED** [review]

File: test/functional/api/vim_spec.lua [hunk 1]
```diff
@@ -178,6 +178,14 @@ describe('API', function()
       -- Verify NO hit-enter prompt.
       eq({mode='n', blocking=false}, nvim("get_mode"))
     end)
+
+    it('Does not cause heap buffer overflow with large output', function()
+      if not os.getenv('ASAN_OPTIONS') then
+        pending('should run with ASAN build', function() end)
+      end
```
Comment (justinmk) [inline test/functional/api/vim_spec.lua]: why not always run it?

Comment (abdelhakeem) [reply test/functional/api/vim_spec.lua]: Thought it would unnecessarily slow down non-ASAN builds, but yeah this is premature optimization (if any) :)

Comment (blueyed) [reply test/functional/api/vim_spec.lua]: Well, it takes quite some time (813ms).
Not much by itself, but this all adds up.

We could maybe have a tag for those kind of slower regression tests?

Comment (blueyed) [reply test/functional/api/vim_spec.lua]: This PR for now is fine for me..!

**Pull request merged**